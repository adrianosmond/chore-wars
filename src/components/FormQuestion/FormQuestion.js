import React from 'react';
import PropTypes from 'prop-types';

const FormQuestion = ({
  id, currentQuestionId, noLabel, error, label, children,
}) => {
  if (id !== currentQuestionId) return null;
  const TagName = noLabel ? 'div' : 'label';
  return (
    <div className="form__question">
      <TagName>
        <span className={`form__label${error ? ' form__label--error' : ''}`}>
          {label}
        </span>
        {children}
      </TagName>
    </div>
  );
};

FormQuestion.propTypes = {
  id: PropTypes.string.isRequired,
  currentQuestionId: PropTypes.string.isRequired,
  noLabel: PropTypes.bool,
  error: PropTypes.bool.isRequired,
  label: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

FormQuestion.defaultProps = {
  noLabel: false,
};

export default FormQuestion;
