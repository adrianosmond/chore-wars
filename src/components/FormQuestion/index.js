import FormQuestion from './FormQuestion';

export default FormQuestion;
