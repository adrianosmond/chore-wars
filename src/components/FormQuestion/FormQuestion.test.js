import React from 'react';
import { shallow } from 'enzyme';
import FormQuestion from './FormQuestion';

describe('FormQuestion', () => {
  const questionChildren = <p id="test-child">Child</p>;
  it('Does not render if the question is not the current question', () => {
    expect(shallow(
      <FormQuestion
        id="test"
        currentQuestionId="not-test"
        error={false}
        label="label"
      />,
    ).html()).toBe(null);
  });

  it('Renders with a label', () => {
    expect(shallow(
      <FormQuestion
        id="test"
        currentQuestionId="test"
        noLabel
        error={false}
        label="label"
      >
        {questionChildren}
      </FormQuestion>,
    )).toMatchSnapshot();
  });

  it('Renders with a div', () => {
    expect(shallow(
      <FormQuestion
        id="test"
        currentQuestionId="test"
        noLabel={false}
        error={false}
        label="label"
      >
        {questionChildren}
      </FormQuestion>,
    )).toMatchSnapshot();
  });

  it('Can display an error', () => {
    expect(shallow(
      <FormQuestion
        id="test"
        currentQuestionId="test"
        noLabel
        error
        label="label"
      >
        {questionChildren}
      </FormQuestion>,
    )).toMatchSnapshot();
  });

  it('Renders children', () => {
    const component = shallow(
      <FormQuestion
        id="test"
        currentQuestionId="test"
        noLabel
        error
        label="label"
      >
        {questionChildren}
      </FormQuestion>,
    );
    const children = component.find('#test-child');
    expect(children.html()).toEqual(shallow(questionChildren).html());
  });
});
