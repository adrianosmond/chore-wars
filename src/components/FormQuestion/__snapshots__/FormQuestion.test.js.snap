// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormQuestion Can display an error 1`] = `
<div
  className="form__question"
>
  <div>
    <span
      className="form__label form__label--error"
    >
      label
    </span>
    <p
      id="test-child"
    >
      Child
    </p>
  </div>
</div>
`;

exports[`FormQuestion Renders with a div 1`] = `
<div
  className="form__question"
>
  <label>
    <span
      className="form__label"
    >
      label
    </span>
    <p
      id="test-child"
    >
      Child
    </p>
  </label>
</div>
`;

exports[`FormQuestion Renders with a label 1`] = `
<div
  className="form__question"
>
  <div>
    <span
      className="form__label"
    >
      label
    </span>
    <p
      id="test-child"
    >
      Child
    </p>
  </div>
</div>
`;
